src/Symfony/Contracts/Service/LazyString.php

```diff
@@ -49,6 +49,47 @@ public static function fromCallable($callback, ...$arguments): self
         return $lazyString;
     }
 
+    /**
+     * @param object|string|int|float|bool $value A scalar or an object that implements the __toString() magic method
+     *
+     * @return static
+     */
+    public static function fromStringable($value): self
+    {
+        if (!self::isStringable($value)) {
+            throw new \TypeError(sprintf('Argument 1 passed to %s() must be a scalar or an object that implements the __toString() magic method, %s given.', __METHOD__, \is_object($value) ? \get_class($value) : \gettype($value)));
+        }
+
+        if (\is_object($value)) {
+            return static::fromCallable([$value, '__toString']);
+        }
+
+        $lazyString = new static();
+        $lazyString->value = (string) $value;
+
+        return $lazyString;
+    }
+
+    /**
+     * Tells whether the provided value can be cast to string.
+     */
+    final public static function isStringable($value): bool
+    {
+        return \is_string($value) || $value instanceof self || (\is_object($value) && method_exists($value, '__toString')) || is_scalar($value);
+    }
+
+    /**
+     * Casts scalars and stringable objects to strings.
+     *
+     * @param object|string|int|float|bool $value
+     *
+     * @throws \TypeError When the provided value is not stringable
+     */
+    final public static function resolve($value): string
```

Comment:
What would be the advantage of using this method over `(string) $value` directly?

Reply:
```
echo (string) [];
```
> Notice: Array to string conversion

Reply:
Got it, thanks!